### ISSUE TITLE
IPX Bad request with firebase storage images (partially encoded url)

### ISSUE THREAD

**Issue by gabrielsze** [nuxt/image#435]
I'm trying to setup nuxt-image for my firebase-hosted images for my static site, and have done the following setup.

Edit: I was able to successfully use nuxt-image with a sample image from unsplash, but it didn't work using my custom images from firebase. The problem seems to be how the image link is handled.

**nuxt.config.js**
```
  target: 'static',
  buildModules: ["vue-notion/nuxt", "@nuxtjs/prismic", '@nuxt/image'],
  image: {
    domains: ['https://firebasestorage.googleapis.com'],
    // adding or removing alias didn't seem to change anything here
    alias: {
      google: 'https://firebasestorage.googleapis.com'
    }
  },
```

I then use nuxt-img here with one of my images

```
<nuxt-img
      class="test"
      src="https://firebasestorage.googleapis.com/v0/b/XXX.appspot.com/o/deals_images%2FKFC%2Fassets%2FKFC_Logo?alt=media&token=5f9afab6-1b8b-41e1-ac2d-83d83cad039c"
      sizes="sm:100px md:200px lg:500px"
    />
```

However, I see this error in the console when i do `npm run dev` and the image is missing.

```
GET http://localhost:3000/_ipx/w_200/https://firebasestorage.googleapis.com/v0/b/XXX.appspot.com/o/deals_images%2FKFC%2Fassets%2FKFC_Logo%3Falt=media%26token=5f9afab6-1b8b-41e1-ac2d-83d83cad039c 400 (IPX: Bad Request)
```

Am I missing anything to use nuxt-image to get responsive images for my static site?

Image link: https://firebasestorage.googleapis.com/v0/b/XXX.appspot.com/o/deals_images%2FKFC%2Fassets%2FKFC_Logo?alt=media&token=5f9afab6-1b8b-41e1-ac2d-83d83cad039c

Edit: Here's another firebase storage image link to try, it still doesn't work for me. Could it be something to do with firebase storage image URLs?

https://firebasestorage.googleapis.com/v0/b/amber-prod.appspot.com/o/images%2Fwatch-dogs-2-sitara-dhawan-render-3-woman-holding-black-and-blue-tumbler-png-clipart_1593206709107?alt=media&token=a42c3f00-89d2-4ce1-9893-da0a5c039f6e

**Comment by gabrielsze** [comment]
Here's a reproduction link https://codesandbox.io/s/interesting-pateu-2so0v?file=/pages/index.vue

**Comment by gabrielsze** [comment]
Hey @shadow81627, that's a nice catch. Once it decodes the entire URL, the image string is no longer valid. That's where i think the bad request happens.

Perhaps a solution would be to check for a firebase storage URL with this format and not decoding it since we have to "treat them as opaque strings that reference the file."

_firebasestorage.googleapis.com/v0/b/{projectId}.appspot.com/o/_

**Comment by wtpalexander** [comment]
Any luck getting this merged @gabrielsze?

**Comment by dosstx** [comment]
I am dealing with this problem now. What is the latest solution, @gabrielsze ?

**Comment by vis97c** [comment]
Using alias, as stated here is the workaround. https://github.com/nuxt/image/issues/392#issuecomment-904457648

That partially fixes the issue. Firebase uses '%2f' for slashes but NuxtImg converts those to real '/' breaking the url mapping

**Comment by alimozdemir** [comment]
I wonder what is the situation here. When I look for alternative implementation on NextJS they are exposing it with encoded url e.g.

```
www.domain.com/images?url=https%3A%2F%2Fgithub.com%2Fnuxt%2Fnuxt.jpg
```

With nuxt image it is e.g.

```
www.domain.com/_ipx/s_320x200/https://github.com/nuxt/nuxt.jpg
```

And whenever we have an encoded thing in the image url we end up with this problem. I think they are doing it with a purpose.

Covering the URL with encodeURIComponent/decodeURIComponent would solve the problem I suppose but I'm not sure where is it handled so I can't raise a PR for it right now. But let me show you what I mean

**there is a `%2f` in the URL**
```
encodeURIComponent("https://github.com/nuxt/nuxt%2f.jpg")
```
output: `https%3A%2F%2Fgithub.com%2Fnuxt%2Fnuxt%252f.jpg`
url: `www.domain.com/_ipx/s_320x200/https%3A%2F%2Fgithub.com%2Fnuxt%2Fnuxt%252f.jpg`

Whenever the lib wants to consume, it should decode it
```
decodeURIComponent('https%3A%2F%2Fgithub.com%2Fnuxt%2Fnuxt%252f.jpg')
```
output: `https://github.com/nuxt/nuxt%2f.jpg`

So we preserved `%2f` while passing the data.

Let me know what do you think @pi0

**Comment by dosstx** [comment]
Yes, that is how I handled my problem with the FIrebase Storage URLs and using Nuxt Img. Here is very basic level for clarity:

```
<template>
    <div>
        <NuxtImg :src="fixedUrl" />
    </div>
</template>

<script setup lang="ts">
const url = ref('https://firebasestorage.googleapis.com/v0/b/bedtimeapp-stories/o/kRUx8ZxLoUd0y7kTcxgPfwTiGu13%2Fstory-images%2FfileNameHere.png?alt=media&token=400246d4-d7ea-4169-a420-e837fefe040a')

const splicedImgUrl = url.value.replace('https://firebasestorage.googleapis.com/v0/b/bedtimeapp-stories/o/', '')

console.log('splicedImgUrl: ', splicedImgUrl)

const encodedImagePath = encodeURIComponent(splicedImgUrl)

console.log('encodedImagePath: ', encodedImagePath)

const fixedUrl = 'https://firebasestorage.googleapis.com/v0/b/bedtimeapp-stories/o/' + encodedImagePath

</script>

**Comment by AsgarovAsgar** [comment]
@dosstx 
I created a test page and then added the same code. But it did not work. Could you share your nuxt.config.ts file, please?

**Comment by anthony32773** [comment]
Has anyone discovered a fix for this issue? @gabrielsze Did you manage to fix this? I am encountering the same issue.

**Comment by richard-ulcuango** [comment]
After multiple tests I was able to replace the special characters that generated conflict because apparently nuxt image formatted them twice.

Now the problem is that nuxt image generates 503 errors to images randomly in its version 1.8.1 there is still no solution to that problem so I reconsider not to use the module until I have a solution to that problem.

<script lang="ts" setup>
defineProps<{
  src: string;
  alt: string;
}>();
function formatUrl(url: string): string {
  const data = new URL(url);
  const href = data.origin + data.pathname + '?alt=media';
 // Replace %20 with an empty space and other special characters according to the response of the nuxt image
  return href.replaceAll('%20', '  ').replaceAll('%2F', '%252F');
}
</script>
<template>
  <NuxtImg
    :alt="alt"
    :title="alt"
    loading="lazy"
    :src="formatUrl(src)"
    sizes="100vw sm:50vw md:400px"
    format="webp"
  />
</template>
